### PR TITLE
Review data providers

### PR DESCRIPTION
**Pull request by machi1990**
Round of database perf analysis as par https://github.com/aerogear/graphback/issues/1758

### CODE REVIEW

File: packages/graphback-runtime-knex/src/KnexDBDataProvider.ts [hunk 1]
```diff
@@ -39,26 +39,21 @@ export class KnexDBDataProvider<Type = any> implements GraphbackDataProvider<Typ
...
-    if (updateResult === 1) {
-      //tslint:disable-next-line: await-promise
-      const dbResult = await this.db.select(this.getSelectedFields(selectedFields)).from(this.tableName).where(idField.name, '=', idField.value);
-      if (dbResult && dbResult[0]) {
-        return dbResult[0]
-      }
+    const updateResult = await this.db(this.tableName).update(updateData, this.getSelectedFields(selectedFields)).where(idField.name, '=', idField.value);
+    if (updateResult && updateResult[0]) {
+      return updateResult[0];
     }
```
Comment (craicoverflow) [inline packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:45]: @wtrocki I'd love to know the original reason for two separate queries in the first place - I recall there being a reason so don't want to approve this without input on this

File: packages/graphback-runtime-knex/src/KnexDBDataProvider.ts [hunk 2]
```diff
@@ -39,26 +39,21 @@ export class KnexDBDataProvider<Type = any> implements GraphbackDataProvider<Typ
...
+
     throw new NoDataError(`Cannot update ${this.tableName}`);
   }
 
   public async delete(data: Partial<Type>, selectedFields?: string[]): Promise<Type> {
     const { idField } = getDatabaseArguments(this.tableMap, data);
 
     //tslint:disable-next-line: await-promise
-    const beforeDelete = await this.db.select(this.getSelectedFields(selectedFields)).from(this.tableName).where(idField.name, '=', idField.value);
```
Comment (craicoverflow) [inline packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:57]: @wtrocki I'd love to know the original reason for two separate queries in the first place - I recall there being a reason so don't want to approve this without input on this

File: packages/graphback-runtime-knex/src/KnexDBDataProvider.ts (hunk 1 above)
Comment (wtrocki) [reply packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:45]: Support for MYSQL, Postgres. Some dbs return different data. For PG this is good

File: packages/graphback-runtime-knex/src/KnexDBDataProvider.ts (hunk 2 above)
Comment (wtrocki) [reply packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:57]: This provider history originates from times where we were supporting sqllite etc.

File: packages/graphback-runtime-knex/src/KnexDBDataProvider.ts (hunk 1 above)
Comment (machi1990) [reply packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:45]: Thanks for the context. But what this patch brings is no different from https://github.com/aerogear/graphback/blob/9615ffb50453b9772b39eda22d645f16dd91569a/packages/graphback-runtime-knex/src/KnexDBDataProvider.ts#L31

To have a more general purpose Relational database data provider, eventually, I'd see us renaming the  [SQLiteKnexDBDataProvider](https://github.com/aerogear/graphback/blob/9615ffb50453b9772b39eda22d645f16dd91569a/packages/graphback-runtime-knex/src/SQLiteKnexDBDataProvider.ts) to something more general and guide people on how to integrate it in their application. Much like what is done here https://graphback.dev/docs/databases/sqlite#using-knex-provider

Comment (wtrocki) [reply packages/graphback-runtime-knex/src/KnexDBDataProvider.ts:45]: Yep. I did not meant to say your change is somehow wrong. We should merge it. eventually check docs for sqlite

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**